tsconfig.json

```diff
@@ -1,21 +1,30 @@
 {
   "compilerOptions": {
+    "moduleDetection": "force",
+    "isolatedModules": true,
+    "allowJs": true,
+    "resolveJsonModule": true,
+    "allowUnreachableCode": false,
+    "forceConsistentCasingInFileNames": true,
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitUseStrict": false,
+    "noStrictGenericChecks": false,
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "importHelpers": true,
+    "skipLibCheck": true,
+    "noEmitOnError": true,
     "allowSyntheticDefaultImports": true,
-    "rootDir": "./src",
-    "baseUrl": "./src",
     "esModuleInterop": true,
-    "moduleResolution": "node",
-    "lib": ["DOM", "ES6"],
-    "noEmitOnError": false,
+    "moduleResolution": "Node",
+    "lib": ["ES6", "DOM"],
     "noImplicitAny": true,
-    "preserveConstEnums": true,
     "strict": true,
     "strictNullChecks": true,
     "declaration": true,
-    "declarationMap": true,
-    "declarationDir": "./dist/types",
-    "module": "commonjs",
-    "target": "ES5"
+    "module": "ES2020",
+    "target": "ES6",
```

Comment:
We should probably review the target too.

Reply:
Yeah, let's use ES2020 for the time being. It aligns with the other JS SDKs.